.github/workflows/ci_plugins.yaml

```diff
@@ -0,0 +1,104 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  prepare_matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get changed files for plugins
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v44
+
+      - name: Filter changed plugins and set output
+        id: set-matrix
+        run: |
+          IFS=$'\n'
+          changed_files=(${{ steps.changed-files-specific.outputs.all_changed_files }})
+          declare -A plugin_set
+          for file in "${changed_files[@]}"; do
+            if [[ "$file" =~ ^plugins/ && ! "$file" =~ ^plugins/template/ ]]; then
+              plugin_name=$(echo "$file" | cut -d '/' -f 2)
+              plugin_set[$plugin_name]=1
+            fi
+          done
+          plugins=("${!plugin_set[@]}")
+          matrix_json=$(printf '%s\n' "${plugins[@]}" | jq -R -s -c '{plugin: split("\n")[:-1]}')
+          echo "matrix_json=$matrix_json"
+          echo "Changed plugins: ${plugins[*]}"
+          echo "matrix=$matrix_json" >> $GITHUB_OUTPUT
+          echo "matrix_json: $matrix_json"
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  test_plugin:
+    needs: prepare_matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.prepare_matrix.outputs.matrix)}}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache Pip Packages
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip' # caching pip dependencies
+
+      - name: Cache Python Installation
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.pythonLocation }} # Cache the whole python installation dir.
+          key: ${{ matrix.plugin }}_${{ hashFiles('pyproject.toml', '*/pyproject.toml') }}
+
+
+      - name: Install SuperDuperDB Project
+        run: |
+          # Install core and testsuite dependencies on the cached python environment.
+          python -m pip install '.[test]'
+
+      - name: Install DevKit (docs, testing, etc)
+        run: |
+          make install_devkit
```

Comment:
Maybe we should combine these 2 things into `.[test]`

Reply:
Yes